### DISCUSSION TITLE
Adjusting to build in a different language

### DISCUSSION THREAD

**Discussion by benbodhi** [AntonOsika/gpt-engineer#228] [General]
I've been experimenting and attempting to adjust gpt-engineer to create complete WordPress plugins (PHP, JS, HTML, CSS). 

I've managed to get it to generate a skeleton architecture without much functionality. Then added a lot more context to the main prompt and fore some reason it wouldn't create the files anymore (the content was in the full response so not a big deal, can create manually).

Will continue experimenting and tweaking to see if I can make this use case possible.

Instead of keeping a note with all my plugin ideas, I'm working to be able to jot an idea down with gpt-engineer and have it build a working mvp.

It's been fun and just wanted to share in case anyone else is doing something similar or wants to chat about getting this to build in other languages and specific styles.

Legendary repo, thanks!

**Comment by echogit** [comment]
I actually thought of this exact use case when I frist encountered gpt-engineer. Please update when you get something working.

**Comment by noxiouscardiumdimidium** [comment]
start your prompt with something like "stat [language] entry files for [project name] [one or more files you want written to disk on first run if possible]"

you might also wanna try my experimental philosophy template. its only been tested on the latest major pip install update, and todays most recent git pull - i have no idea how it might react to an out of date build attempt... i dont advise that! xD

*** take into account this is NOT meant to spit out functional software in a single build attempt, and some steps MUST output in python for the early build iterations to produce actionable error messages during function testing steps, it figures out how to break it in python***

add a new top line "Always prefers [lang] over python for final build, when entry file is [extension]"

```
Apply extraction and inversion, whenever possible, to condense code, merge functions, and streamline debugging (i.e. "never-nesting").
Avoid needlessly complicated lamdba functions, you're bad at it.
Keep different classes in separate files to maintain code organization and modularity.
For Python projects, always include an appropriate requirements.txt file to specify the required dependencies.
For Node.js projects, always create an appropriate package.json file to manage project dependencies and configurations.
Add comments to briefly describe the purpose and functionality of each function definition.
Consider adding comments to explain complex sections of logic, making the code more understandable and maintainable.
Adhere to the best practices for the chosen programming languages, following industry conventions and style guides.
Remember, these guidelines can serve as a starting point, and you can always adapt and customize them based on your project's specific requirements and team preferences.

Python toolbelt preferences:
-Utilize pytest for unit testing and test automation.
-Leverage dataclasses to create structured and immutable data objects efficiently.
```

**Reply by benbodhi** [reply]
This is almost exactly what I've done so far in terms of prompting. I haven't actually let it write in python first and then rewrite in php for the build. But I'll give that a go and see if it will create functionality in all the methods.
